### DISCUSSION TITLE
refetchIntervalInBackground behaviour on mobile

### DISCUSSION THREAD

**Discussion by ikryvorotenko** [TanStack/query#5170] [Q&A] [answered]
I'm trying to understand what would be the behavior of `refetchIntervalInBackground` flag on mobile devices. Like when the user switches the browser app or locks the phone. 

As I understand, when the browser is not active, `setInterval` is not executed for refetch. Considering that, does it make more sense to have it implemented via timestamp in the state and custom `refetchOnWindowFocus`, like 
```
const TTL = 20_000

useQuery({...
      refetchInterval: TTL,
      refetchOnWindowFocus: query => {
          const timestamp = query.state.data.timestamp
          const shouldRefetch = Date.now() > (timestamp + TTL);
          return shouldRefetch
      }
 }),
```

**Comment by TkDodo** [comment, marked as answer]
`refetchIntervalInBackground` checks for `focusManager.isFocused`, which you might need to implement differently in react-native: https://tanstack.com/query/v4/docs/react/react-native#refetch-on-app-focus

If the question wasn't about react native, but just browsers on mobile: `focusManager` is implemented with the [visibilitychange event](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event), so I think it'll work just the same as on a desktop
